python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -169,7 +169,7 @@ def _from_host_memory(cls: Type[T], data: Any) -> T:
         # Create from device memory
         return cls._from_device_memory(buf)
 
-    def _getitem(self, offset: int, size: int) -> Buffer:
+    def _getitem(self: T, offset: int, size: int) -> T:
```

Comment:
Throughout, use `from typing_extensions import Self` and then
```suggestion
    def _getitem(self: Self, offset: int, size: int) -> Self:
```

ref: (PEP 673)[https://peps.python.org/pep-0673/]

Reply:
I guess we need `mypy v1.0.0` for `Self` support: https://github.com/python/mypy/issues/11871
I am getting:
```
python/cudf/cudf/core/buffer/buffer.py:173: error: Variable "typing_extensions.Self" is not valid as a type
python/cudf/cudf/core/buffer/buffer.py:173: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
python/cudf/cudf/core/buffer/buffer.py:186: error: Variable "typing_extensions.Self" is not valid as a type
python/cudf/cudf/core/buffer/buffer.py:186: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
python/cudf/cudf/core/buffer/buffer.py:198: error: Variable "typing_extensions.Self" is not valid as a type
python/cudf/cudf/core/buffer/buffer.py:198: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
python/cudf/cudf/core/buffer/tenable_buffer.py:262: error: Value of type Self? is not indexable
Found 4 errors in 2 files (checked 149 source files)
```

Reply:
Maybe submit a small PR that replaces all `self: T` and bump the mypy version?

Reply:
If you merge trunk you can now do this, I opened https://github.com/rapidsai/cudf/pull/13346 for the other cases.